Feature/smart field types

Obversity opened the pull request:
Additions to the  koi controller generators to automatically generate configuration/validation for various common field types we use, including enums for type/status pulldowns, dragonfly images/files, associations, and url fields, dates and booleans.

See RFC: #236

Obversity committed:
added smarter field types for images/files/enums in generators

Obversity committed:
added field config to all generator model templates, added permitted params for files

Obversity committed:
formatting fixes and making new generator code more readable

Obversity committed:
added belongs_to associations to smart field types in controller generators

pvawser01 commented on lib/generators/koi/mailer/templates/model_template.rb:
```diff
@@ -2,10 +2,29 @@ class <%= class_name %> < ActiveRecord::Base
   has_crud settings: true
 
+  <%- model_attributes.each do |attr| -%>
+      <%- if field_is_enum?(attr.name, attr.type) -%>
```
Spacing is all over the place here. Can the block content not be replaced with <%= make_field_config(attr) -%>

pvawser01 commented:
See comment

Obversity committed:
fixes to formatting and guessing of field types / config in generators

Obversity committed:
updates to smarter generators
- accounted for polymorphic and indexed fields
- added some default validation for booleans
- refactored slightly

Obversity committed:
fixed attachment image helper so that it doesn't crash when provided the wrong file type

Obversity committed:
fixed file-type guessing method in generators to catch all _uid fields that the image guessing method doesn't

Obversity committed:
various fixes / improvements  to mailer / controller generators
